File: easy/can_construct_ransom_note.go
package easy

import "strings"

func CanConstructRansomNote(ransomNote string, magazine string) bool {
	const aCode = 'a'
	var freq [26]int

	// Count letters in magazine (case-insensitive)
	for _, ch := range strings.ToLower(magazine) {
		if ch >= 'a' && ch <= 'z' {
			freq[ch-aCode]++
		}
	}

	// Subtract letters used by ransomNote
	for _, ch := range strings.ToLower(ransomNote) {
		if ch >= 'a' && ch <= 'z' {
			freq[ch-aCode]--
			if freq[ch-aCode] < 0 {
				return false
			}
		}
	}
	return true
}

File: easy/can_construct_ransom_note.py
from collections import Counter


def can_construct_ransom_note(ransom_note: str, magazine: str) -> bool:
    freq = Counter(magazine.lower())
    for char in ransom_note:
        ch = char.lower()
        if ch not in freq:
            return False
        count = freq.get(ch, 0)
        if count < 1:
            return False
        freq[ch] = count - 1
    return True
